modules/photo/src/ccm/utils.cpp

```diff
@@ -0,0 +1,86 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Author: Longbu Wang <wanglongbu@huawei.com.com>
+//         Jinheng Zhang <zhangjinheng1@huawei.com>
+//         Chenqi Shan <shanchenqi@huawei.com>
+
+#include "utils.hpp"
+
+namespace cv {
+namespace ccm {
+
+inline double gammaCorrection_(double element, double gamma)
+{
+    return (element >= 0 ? pow(element, gamma) : -pow((-element), gamma));
+}
+
+Mat gammaCorrection(const Mat& src, double gamma, Mat dst)
+{
+    return elementWise(src, [gamma](double element) -> double { return gammaCorrection_(element, gamma); }, dst);
+}
```

Comment:
I propose to convert Gamma correction to public function with InputArray and OutputArray. It's useful independently from the pipeline and also may be significantly optimized with universal intrinsics.